### ISSUE TITLE
How could I know the label is correctly read from image_dataset_from_directory?

### ISSUE THREAD

**Issue by laurence-lin** [tensorflow/tensorflow#46999]
I'd save the directory structure as indicated in document, separate classes of images stored in corresponding folder with corresponding label as folder name.

However, I could not find a way to return the labels read from image_dataset_from_directory function, and I'm afraid the class name is not correctly read.

The performance of my training is that, after 10 training epochs the training and validate accuracy reached 0.99, but the model prediction would only predict one class.

How could I check the label read by this function?

**Comment by ravikyram** [comment]
@laurence-lin 

This question is better asked on StackOverflow since it is not a bug or feature request. There is also a larger community that reads questions there and provide better and faster support for such issues. Thanks!

**Comment by laurence-lin** [comment]
> @laurence-lin
> 
> This question is better asked on StackOverflow since it is not a bug or feature request. There is also a larger community that reads questions there and provide better and faster support for such issues. Thanks!

Alright, because I couldn't find the solution on stackoverflow. Thanks!

**Comment by ymodak** [comment]
You can find the labels in the `class_names` attribute on the datasets created by `image_dataset_from_directory` .

If your directory structure is like;
```python
  main_directory/
  ...class_a/
  ......a_image_1.jpg
  ......a_image_2.jpg
  ...class_b/
  ......b_image_1.jpg
  ......b_image_2.jpg
  ```

```python
train_ds = tf.keras.preprocessing.image_dataset_from_directory(data_dir, validation_split=0.2, 
                  subset="training", seed=123,  
                  image_size=(img_height, img_width), batch_size=batch_size)
# It outputs something like:
Found X files belonging to 2 classes.
Using X files for training.
```
These classes correspond to the directory names in alphabetical order and can be inferred as,
```python
class_names = train_ds.class_names
print(class_names)
# class_a, class_b
```
